### ISSUE TITLE
Graph::ListGroups API call fails with a JSON unmarshaling error

### ISSUE THREAD

**Issue by nmiodice** [microsoft/azure-devops-go-api#30]
The Graph API implementation for `ListGroups` consistently fails with a JSON unmarshaling error.

Steps to reproduce:
```
... initialize client
_, err := client.ListGroups(clients.ctx, graph.ListGroupsArgs{})
```

In my project, I need to call the API like this, which also fails:
```
client.ListGroups(clients.ctx, graph.ListGroupsArgs{
		ScopeDescriptor: &projectDescriptor,
		ContinuationToken: continuationToken,
})
```

After a brief look into the code, my initial hunch is that, possibly, the unmarshal code being called is `UnmarshalBody`, where it may need to be `UnmarshalCollectionBody`

The response in browser looks like this
```json
{
  "count": 10,
  "value": [
    ...
  ]
}
```

**Comment by tiwood** [comment]
I've just hit the same issue with Graph::ListUsers.
This may affect all `List` methods probably?

**Comment by tedchamb** [comment]
Fixed in PR #31
